### PR TITLE
[Form] properly parse dates before the Gregorian calendar

### PR DESCRIPTION
**Pull request by xabbuh**
| Q             | A
| ------------- | ---
| Branch?       | 3.4
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #29610 #31057
| License       | MIT
| Doc PR        |

### CODE REVIEW

File: src/Symfony/Component/Form/Extension/Core/DataTransformer/DateTimeToLocalizedStringTransformer.php [hunk 1]
```diff
@@ -199,4 +200,12 @@ protected function isPatternDateOnly()
         // check for the absence of time-related placeholders
         return 0 === preg_match('#[ahHkKmsSAzZOvVxX]#', $pattern);
     }
+
+    private function getConverterFormatter()
+    {
+        // see https://bugs.php.net/bug.php?id=66323
+        $timezone = class_exists('IntlTimeZone', false) ? \IntlTimeZone::createTimeZone('UTC') : 'UTC';
```
Comment (ro0NL) [inline src/Symfony/Component/Form/Extension/Core/DataTransformer/DateTimeToLocalizedStringTransformer.php]: we can already past timezone as a string :/ the bug report is for using integers (e.g. derived from DateTimeZone::EUROPE) which is not supported here.

File: src/Symfony/Component/Form/Extension/Core/DataTransformer/DateTimeToLocalizedStringTransformer.php [hunk 2]
```diff
@@ -199,4 +200,16 @@ protected function isPatternDateOnly()
         // check for the absence of time-related placeholders
         return 0 === preg_match('#[ahHkKmsSAzZOvVxX]#', $pattern);
     }
+
+    private function getCalendarConvertingFormatter()
+    {
+        $timezone = $this->isPatternDateOnly() ? 'UTC' : $this->inputTimezone;
+
+        if (class_exists('IntlTimeZone', false)) {
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Form/Extension/Core/DataTransformer/DateTimeToLocalizedStringTransformer.php]: shouldn't we cache the result of this call - or the result of the method?

**Review by nicolas-grekas — DISMISSED** [review]

**Comment by nicolas-grekas** [comment]
Status: needs work
(tests fail on appveyor)

**Comment by javiereguiluz** [comment]
Could any Windows user please help us debug the failing test? Thanks.

**Comment by nicolas-grekas** [comment]
Could the failure be related to an old version of intl?

**Comment by ro0NL** [comment]
i wasnt able to reproduce on ICU 52.1 on Linux

```
php -r "var_dump((new DateTime((new IntlDateFormatter('en', 2, -1, new DateTimeZone('UTC'), 1, 'yyyy-MM-dd HH:mm:ss'))->format(-32157216000)))->format('Y-m-d'));"
string(10) "0950-12-19"
```

**Comment by xabbuh** [comment]
closing here as I don't know how to make tests succeed on AppVeyor

**Pull request closed without merging**